provider/testdata/src/lib.rs

```diff
@@ -222,17 +222,17 @@ pub fn buffer_no_fallback() -> impl BufferProvider {
 #[cfg(feature = "buffer")]
 pub fn small_buffer() -> impl BufferProvider {
     lazy_static::lazy_static! {
-        static ref SMALLER_POSTCARD: icu_provider_blob::StaticDataProvider = {
+        static ref SMALLER_POSTCARD: icu_provider_blob::BlobDataProvider = {
             // The statically compiled data file is valid.
             #[allow(clippy::unwrap_used)]
-            icu_provider_blob::StaticDataProvider::try_new_from_static_blob(include_bytes!(concat!(
+            icu_provider_blob::BlobDataProvider::try_new_from_static_blob(include_bytes!(concat!(
                 env!("CARGO_MANIFEST_DIR"),
                 "/data/decimal-bn-en.postcard"
             )))
             .unwrap()
         };
     }
-    *SMALLER_POSTCARD
+    SMALLER_POSTCARD.clone()
```

Comment:
Issue: There is a slight code size hit unfortunately

Old:

```
-rwxr-xr-x 1 runner docker    44072 Sep 20 15:03 optim4.elf
-rwxr-xr-x 1 runner docker    26824 Sep 20 15:03 optim5.elf
```

New:

```
-rwxr-xr-x 1 runner docker    44640 Sep 20 13:56 optim4.elf
-rwxr-xr-x 1 runner docker    27144 Sep 20 13:56 optim5.elf
```

I would rather not accept code size regressions for changes that are refactoring only without a value add.

As a reminder, these can be reproduced locally with `cargo make test-c-tiny`

If `small_buffer()` were migrated over to databake, this issue might become moot.

Reply:
Why are we code-size optimising testdata?

Reply:
We're optimizing the code size of the data provider.

Reply:
The problem is that we're not optimizing testdata for size, we're optimizing it for performance. That's why we have a `lazy_static` that we clone, and that plus the cloning costs code size.

I've moved the small buffer out of testdata and into the tiny-c test. PTAL